charts/dns/templates/pdns-deployment.yaml

```
@@ -6,65 +6,125 @@ metadata:
+        {{- with .Values.podAnnotations }}
         {{- toYaml . | nindent 8 }}
-      {{- end }}
+        {{- end }}
       labels:
+        {{- include "stateless-dns.labels" . | nindent 8 }}
         {{- include "pdns.selectorLabels" . | nindent 8 }}
     spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "stateless-dns.serviceAccountName" . }}
+      {{- with .Values.shareProcessNamespace }}
+      shareProcessNamespace: {{ . }}
+      {{- end }}
+      {{- with .Values.podSecurityContext }}
+      securityContext:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.priorityClassName }}
+      priorityClassName: {{ . | quote }}
+      {{- end }}
+      {{- with .Values.dnsPolicy }}
+      dnsPolicy: {{ . }}
+      {{- end }}
       containers:
         - name: pdns
-          image: "{{ .Values.pdns.repository }}:{{ .Values.pdns.tag }}"
+          image: "{{ .Values.pdns.registry }}/{{ .Values.pdns.repository }}:{{ .Values.pdns.tag }}"
           imagePullPolicy: {{ .Values.pdns.pullPolicy }}
+          {{- with .Values.pdns.securityContext }}
           securityContext:
-            allowPrivilegeEscalation: false
-            capabilities:
-              add:
-              - NET_BIND_SERVICE
-              drop:
-              - all
-          resources:
-            {{- toYaml .Values.pdns.resources | nindent 12 }}
-          {{- if or .Values.pdns.httpService.enabled (or .Values.pdns.udpService.enabled .Values.pdns.tcpService.enabled) }}
-          ports:
-          {{- if .Values.pdns.httpService.enabled }}
-          - containerPort: {{ .Values.pdns.httpService.port }}
-            name: http
-            protocol: TCP
-          {{- end }}
-          {{- if .Values.pdns.udpService.enabled }}
-          - containerPort: {{ .Values.pdns.udpService.port }}
-            name: dns-udp
-            protocol: UDP
-          {{- end }}
-          {{- if .Values.pdns.tcpService.enabled }}
-          - containerPort: {{ .Values.pdns.tcpService.port }}
-            name: dns-tcp
-            protocol: TCP
+            {{- toYaml . | nindent 12 }}
           {{- end }}
+          command:
+            - /bin/bash
+            - /entrypoint/script
+            - /usr/local/sbin/pdns_server
+          args:
+            - --launch=gsqlite3
+            - --gsqlite3-database=$(DATABASE_PATH)
+
+            {{- if .Values.service.http.enabled }}
+            - --webserver=yes
+            - --webserver-address=0.0.0.0
+            - --webserver-port={{ .Values.service.http.port }}
+            - --api=yes
+            - --api-key=$(POWERDNS_API_KEY)
+            {{- end }}
+
+            {{- range $arg := .Values.pdns.extraArgs }}
+            - {{ $arg }}
+            {{- end }}
+          {{- if or .Values.service.http.enabled (or .Values.service.udp.enabled .Values.service.tcp.enabled) }}
```

Haven't checked but I think this can be simplified as:

```suggestion
          {{- if or .Values.service.http.enabled.Values.service.udp.enabled .Values.service.tcp.enabled }}
```

---

Yeah, you are right. It was a copy and paste from the old chart and I did not realize of it.

Removed.